python/tank/platform/engine.py

```diff
@@ -1315,11 +1378,23 @@ def __execute_in_main_thread(self, fn):
     ##########################################################################################
     # private         
         
-    def __load_apps(self):
+    def __load_apps(self, reuse_existing_apps=False):
         """
         Populate the __applications dictionary, skip over apps that fail to initialize.
         """
+        # If this is a load as part of a context change, the applications
+        # dict will already have stuff in it. We can explicitly clean that
+        # out here since those apps also exist in self.__application_pool,
+        # which is persistent.
+        self.__applications = dict()
+
         for app_instance_name in self.__env.get_apps(self.__engine_instance_name):
+            # If we're told to reuse existing app instances, check for it and
+            # continue if it's already there. This is most likely a context
+            # change that's in progress, which means we only want to load apps
+            # that aren't already up and running.
+            if reuse_existing_apps and app_instance_name in self.__application_pool:
+                self.__applications[app_instance_name] = self.__application_pool[app_instance_name]
```

Comment:
looking at the code, feels like an else or continue is missing here? Should it not just skip further loading in case the app is already pooled? Sorry if i'm overlooking something obvious! :)